src/Groups/GrpAb.jl

```diff
@@ -365,13 +386,31 @@ function abelian_invariants_of_vector(::Type{T}, v::Vector) where T <: IntegerUn
   return sort!(invs)
 end
 
-# Let `v` be a vector of integers.
-# This function returns the unique vector `w` of nonnegative integers
-# such that `w[1] != 1`, `w[i]` divides `w[i+1]` for `1 < i < length(w)-1`
-# and such that `v` and `w` describe the same abelian group in the sense that
-# the direct product of the groups `ZZ /(v[i]*ZZ)` is isomorphic to
-# the direct product of the groups `ZZ /(w[i]*ZZ)`.
-function elementary_divisors_of_vector(::Type{T}, v::Vector) where T <: IntegerUnion
+"""
+    elementary_divisors(::Type{T} = S, v::Vector{S}) where {T <: IntegerUnion, S <: IntegerUnion}
+
+Return the unique vector `w` of nonnegative integers of type `T`
+such that `w[1] != 1`, `w[i]` divides `w[i+1]` for `1 < i < length(w)-1`
+and such that `v` and `w` describe the same abelian group in the sense that
+the direct product of the groups `ZZ /(v[i]*ZZ)` is isomorphic to
+the direct product of the groups `ZZ /(w[i]*ZZ)`.
```

Comment:
Hmmm... 
```
julia> ZZ/(5*ZZ)
ERROR: MethodError: no method matching /(::ZZRing, ::Hecke.ZZIdl)
```
and if it worked, it'd be a ring not a gorup.

I think we either need to spell this out ("... cyclic groups of order `v[i]`" or so), or else use math mode ($\mathbb{Z}/v_i\mathbb{Z}$ or so)